lib/child_process.js

```diff
@@ -121,29 +121,24 @@ exports._forkChild = function(fd) {
 };
 
 
-function normalizeExecArgs(command, options, callback) {
+function normalizeExecArgs(file, options, callback) {
   if (typeof options === 'function') {
     callback = options;
-    options = undefined;
+    options = {};
+  } else {
+    // Make a shallow copy so we don't clobber the user's options object.
+    options = util._extend({}, options);
```

Comment:
Might just be me but I would prefer less `util._extend`, not more, haha... especially in code that isn't hot path.